### DISCUSSION TITLE
Possible to modify Device Group MeshID?

### DISCUSSION THREAD

**Discussion by PathfinderNetworks** [Ylianst/MeshCentral#3713] [Q&A] [answered]
Here's the scenario - my MeshCentral virtual machine had some issues and I ended up having to roll back quite a bit (nothing to do with MeshCentral itself).  As a result I lost a newer Device Group and its associated devices since the roll back was a date before that device group had existed.  
About a month had gone by before I realized that device group (and it's  devices) were missing- so I really don't want to try and just do a database restore since that will impact the existing data for all of my other groups and devices.  
I re-created the missing Device Group but, obviously, the devices that were assigned to that group won't connect to it due to the MeshID being different (and I'm hoping to avoid reinstall the MeshAgent on all of the devices).
Is there any way possible to modify the MeshID of the 'new' device group to match that of the original device group?  And, even if that is possible, will doing so allow all the devices that have the MeshAgent from the 'old' device group reconnect to the new one?  

Hopefully that scenario makes sense.

**Comment by JSuenram** [comment]
I believe it could be changed directly in the Database, but if that is really a good idea... unsure....

To keep it the supported way, I would recommend to reinstall the agents.... sorry...

**Comment by Ylianst** [comment, marked as answer]
Sorry to be late to answer this, try adding this in the domain section of the config.json:

```
"orphanAgentUser": "admin"
```

There "admin" is the userid of a account in the domain. Once set, any device that connects with a meshid that is unknown will automatically have a device group created under the specified account and the devices will join it. The new device group will be named after the meshid, so you will have to rename it to sometime more friendly and assign the right owner to the device group.

This option has been used in the past for server recovery. You can even launch a server with a completely new database and all the devices will reconnect.

Hope that helps,
Ylian
